components/account/AccountButton.tsx

```
@@ -163,262 +229,413 @@ const AccountButton: FC<{
-                        className={`relative z-30 flex	h-full flex-1 cursor-pointer items-center justify-end rounded-full  ${open
-                          ? "border-orange-500"
-                          : pathname === "/"
-                            ? " border-white"
-                            : "border-black"
-                          }`}
-                      >
-                        <div
-                          className={`flex h-full items-center rounded-full border-2 bg-black py-1 pl-1.5 text-white transition-all md:py-0 ${open ? "border-sunglow-2" : "border-white"
+            {({ open, close }) => {
+              // Track menu open state for scroll locking
+              useEffect(() => {
```

Biome (2.1.2)

[error] This hook is being called from a nested function, but all hooks must be called unconditionally from the top-level component.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)